block-node/block-providers/files.historic/src/test/java/org/hiero/block/node/blocks/files/historic/BlockFileHistoricPluginTest.java

```diff
@@ -984,4 +984,56 @@ void testRetentionPolicyThresholdDisabled() throws IOException {
             }
         }
     }
+
+    /**
+     * Regression tests.
+     */
+    @Nested
+    @DisplayName("Regression Tests")
+    final class RegressionTests extends PluginTestBase<BlockFileHistoricPlugin, BlockingExecutor> {
+        /** The test historical block facility scoped to this regression scenario. */
+        private final SimpleInMemoryHistoricalBlockFacility regressionHistoricalBlockFacility =
+                new SimpleInMemoryHistoricalBlockFacility();
+
+        RegressionTests() {
+            super(new BlockingExecutor(new LinkedBlockingQueue<>()));
+        }
+
+        private Map<String, String> buildConfigOverrides(final FilesHistoricConfig config) {
+            final Entry<String, String> rootPath =
+                    Map.entry("files.historic.rootPath", config.rootPath().toString());
+            final Entry<String, String> compression =
+                    Map.entry("files.historic.compression", config.compression().name());
+            final Entry<String, String> powersOfTenPerZipFileContents = Map.entry(
+                    "files.historic.powersOfTenPerZipFileContents",
+                    String.valueOf(config.powersOfTenPerZipFileContents()));
+            final Entry<String, String> blockRetentionThreshold = Map.entry(
+                    "files.historic.blockRetentionThreshold", String.valueOf(config.blockRetentionThreshold()));
+            return Map.ofEntries(rootPath, compression, powersOfTenPerZipFileContents, blockRetentionThreshold);
+        }
+
+        @Test
+        @DisplayName("init moves corrupted zip file without shutting down")
+        void initDeletesCorruptedZipWithoutShutdown() throws IOException {
```

Comment:
Probably should change the name here to match description.
```suggestion
        void initMovesCorruptedZipWithoutShutdown() throws IOException {
```